distributed/comm/utils.py

```diff
@@ -37,7 +38,12 @@ def _to_frames():
             logger.exception(e)
             raise
 
-    if FRAME_OFFLOAD_THRESHOLD and sizeof(msg) > FRAME_OFFLOAD_THRESHOLD:
+    try:
+        msg_size = sizeof(msg)
+    except RecursionError:
+        msg_size = sys.getsizeof(msg)
```

Comment:
I suggest instead that we default to offloading the serialization work in this case.  This object has proven that it's odd, so it probably makes sense to move it off to another thread for safety.

```suggestion
        msg_size = math.inf
```